lib/index.js

```diff
@@ -77,6 +77,28 @@ class Client {
     return ActionApi.action(this.api, this.apiKey, actionType, data)
   }
 
+  /**
+   * Parameters for get package request
+   * @param {string} id - Id or name of the package
+   * @param {boolean} useDefaultSchema - Indicates whether to use default package schema instead of a custom schema defined with an IDatasetForm plugin (default: false)
+   * @param {boolean} includeTracking - Indicates whether to add tracking information to dataset and resources (default: False)
+   */
+  async retrieve(id, useDefaultSchema = false, includeTracking = false) {
+    const response = await this.action(
+      'package_show',
+      {
+        id,
+        useDefaultSchema,
+        includeTracking,
+      },
+      true
+    )
+
+    const result = mapper.packageCkanToFrictionless(response.result)
+    response.result = result
+    return response
```

Comment:
Why are we returning the response rather than the result. This method should just return the dataset object.

So we can improve could you clarify what was unclear (or could be better) about acceptance criteria in the issue #7:

> client.retrieve(datasetNameOrId) returning a frictionless formatted dataset object.

Reply:
The criteria is clear, just haven't read it carefully